pkg/allocateip/allocateip.go

```diff
@@ -326,21 +487,38 @@ func removeHostTunnelAddr(ctx context.Context, c client.Interface, nodename stri
 		}
 
 		// Determine if we need to do any work.
-		ipipTunnelAddrExists := (node.Spec.BGP != nil && node.Spec.BGP.IPv4IPIPTunnelAddr != "")
+		ipipTunnelAddrExists := node.Spec.BGP != nil && node.Spec.BGP.IPv4IPIPTunnelAddr != ""
 		vxlanTunnelAddrExists := node.Spec.IPv4VXLANTunnelAddr != ""
-		if (vxlan && !vxlanTunnelAddrExists) || (!vxlan && !ipipTunnelAddrExists) {
```

Comment:
@caseydavenport  - this bit of code would have ensured the IP address was assigned a value.  So I figured that was the intent.

My additional check below really is a back up in case it doesn't parse as an IP - but I don't think I've changed the intent.

Reply:
But then take your point that there was probably a bug there.